__tests__/Unit/hooks/tasksApi.test.tsx

```diff
@@ -0,0 +1,212 @@
+                wrapper: Wrapper,
+            }
+        );
+        const [addTask, initialResponse] = result.current;
+        expect(initialResponse.isLoading).toBe(false);
+        expect(initialResponse.data).toBeUndefined();
+
+        act(() => {
+            addTask({
+                type: 'feature',
+                status: 'AVAILABLE',
+                percentCompleted: 0,
+            });
+        });
+
+        await act(() => waitForNextUpdate());
+
+        const nextResponse = result.current[1];
+        expect(nextResponse.isError).toBe(true);
+        expect(nextResponse.error).toHaveProperty('status', 404);
+        expect(nextResponse.error).toHaveProperty('data', {
+            statusCode: 400,
+            error: 'Bad Request',
+            message: '"title" is required',
+        });
+    });
+});
+
+describe('useUpdateTaskMutation()', () => {
+    const updatedTaskData = {
+        status: 'COMPLETED',
+    };
+
+    test('updates a task', async () => {
+        const taskId = '1eJhUW19D556AhPEpdPr';
+        const { result, waitForNextUpdate } = renderHook(
+            () => useUpdateTaskMutation(),
+            {
+                wrapper: Wrapper,
+            }
+        );
+        const [updateTask, initialResponse] = result.current;
+        expect(initialResponse.isLoading).toBe(false);
+        expect(initialResponse.data).toBeUndefined();
+
+        act(() => {
+            updateTask({
+                task: updatedTaskData,
+                id: taskId,
+            });
+        });
+
+        const loadingResponse = result.current[1];
+        expect(loadingResponse.isLoading).toBe(true);
+
+        await waitForNextUpdate({ timeout: 7000 });
+
+        const nextResponse = result.current[1];
+        expect(nextResponse.data).toBeNull();
+        expect(nextResponse.isSuccess).toBe(true);
+        expect(nextResponse.isLoading).toBe(false);
+        expect(nextResponse.status).toBe(QueryStatus.fulfilled);
+    }, 7000);
+
+    test('should fail to update a task with error when invalid id is passed', async () => {
+        server.use(failedUpdateTaskHandler);
+        const { result, waitForNextUpdate } = renderHook(
+            () => useUpdateTaskMutation(),
+            {
+                wrapper: Wrapper,
+            }
+        );
+        const [updateTask, initialResponse] = result.current;
+        expect(initialResponse.isLoading).toBe(false);
+        expect(initialResponse.data).toBeUndefined();
+
+        act(() => {
+            updateTask({ task: updatedTaskData, id: 'incorrectId' });
+        });
+
+        await act(() => waitForNextUpdate());
+
+        const nextResponse = result.current[1];
+        expect(nextResponse.isError).toBe(true);
+        expect(nextResponse.error).toHaveProperty('status', 404);
+        expect(nextResponse.error).toHaveProperty('data', {
+            statusCode: 404,
+            error: 'Not Found',
+            message: 'Task not found',
+        });
```

Comment:
these errors can be stored in fixtures with so it becomes easy for other developers while testing.

Reply:
Shall take this up in a separate PR. Issue has been created.

Reply:
https://github.com/Real-Dev-Squad/website-status/issues/652